src/server/scripts/Events/hallows_end.cpp

```diff
@@ -57,13 +83,452 @@ class spell_hallows_end_has_water_bucket : public SpellScriptLoader
+    {
+        PrepareAuraScript(spell_hallows_end_base_fire_AuraScript);
+
+        void HandleEffectPeriodicUpdate(AuraEffect* aurEff)
+        {
+            if (Unit* owner = GetUnitOwner())
+            {
+                if (Aura* aura = owner->GetAura(SPELL_SMALL_FIRE))
+                    if (aura->GetStackAmount() < 20)
+                        owner->GetAI()->DoCast(SPELL_SMALL_FIRE);
+
+                if (Aura* aura = owner->GetAura(SPELL_BIG_FIRE))
+                    if (aura->GetStackAmount() < 20)
+                        owner->GetAI()->DoCast(SPELL_BIG_FIRE);
+            }
+       }
+
+        void HandleEffectApply(AuraEffect const* aurEff, AuraEffectHandleModes /*mode*/)
+        {
+            if (Unit* owner = GetUnitOwner())
+                if (roll_chance_i(50))
+                    owner->SetAuraStack(SPELL_BIG_FIRE, owner, 20);
+                else
+                    owner->SetAuraStack(SPELL_SMALL_FIRE, owner, 20);
+        }
+
+        void Register() override
+        {
+            OnEffectUpdatePeriodic += AuraEffectUpdatePeriodicFn(spell_hallows_end_base_fire_AuraScript::HandleEffectPeriodicUpdate, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY);
+            OnEffectApply += AuraEffectApplyFn(spell_hallows_end_base_fire_AuraScript::HandleEffectApply, EFFECT_0, SPELL_AURA_PERIODIC_DUMMY, AURA_EFFECT_HANDLE_REAL);
+        }
+    };
+
+    AuraScript* GetAuraScript() const override
+    {
+        return new spell_hallows_end_base_fire_AuraScript();
+    }
+};
+
+class npc_costumed_orphan_matron : public CreatureScript
+{
+public:
+    npc_costumed_orphan_matron() : CreatureScript("npc_costumed_orphan_matron") { }
+
+    struct npc_costumed_orphan_matronAI : public ScriptedAI
+    {
+        npc_costumed_orphan_matronAI(Creature* creature) : ScriptedAI(creature), eventStarted(false) {}
+
+        bool eventStarted;
+        EventMap _events;
+
+        void Reset()
+        {
+            _events.ScheduleEvent(EVENT_BEGIN, /*Minutes(30)*/Seconds(5));
+        }
+
+        void GetInitXYZ(float &x, float &y, float &z, float &o, uint32 &path)
+        {
+            switch (me->GetAreaId())
+            {
+                case 87: // Goldshire
+                    x = -9494.4f; y = 48.53f; z = 70.5f; o = 0.5f; path = 235431;
+                    break;
+                case 131: // Kharanos
+                    x = -5558.34f; y = -499.46f; z = 414.12f; o = 2.08f; path = 235432;
+                    break;
+                case 3576: // Azure Watch
+                    x = -4163.58f; y = -12460.30f; z = 63.02f; o = 4.31f; path = 235433;
+                    break;
+                case 362: // Razor Hill
+                    x = 373.2f; y = -4723.4f; z = 31.2f; o = 3.2f; path = 235434;
+                    break;
+                case 159: // Brill
+                    x = 2195.2f; y = 264.0f; z = 55.62f; o = 0.15f; path = 235435;
+                    break;
+                case 3665: // Falcon Wing Square
```

Comment:
[Google] Did you mean: [**_Falconwing_** Square](https://www.google.com/?gfe_rd=cr&ei=5rIKWNueI5Or8wftjZ3gBg&gws_rd=cr&fg=1#newwindow=1&q=Falcon+Wing+Square)